chart/templates/_helpers.yaml

```
@@ -223,27 +223,28 @@ If release name contains chart name it will be used as a full name.
     - name: GITSYNC_SSH_KNOWN_HOSTS
       value: "false"
     {{- end }}
-    {{ else if .Values.dags.gitSync.credentialsSecret }}
-    - name: GIT_SYNC_USERNAME
+    {{- else if and .Values.dags.gitSync.credentialsSecret (semverCompare ">=v4.0.0" .Values.images.gitSync.tag)}}
+    - name: GITSYNC_USERNAME
       valueFrom:
         secretKeyRef:
           name: {{ .Values.dags.gitSync.credentialsSecret | quote }}
-          key: GIT_SYNC_USERNAME
-    - name: GITSYNC_USERNAME
+          key: GITSYNC_USERNAME
+    - name: GITSYNC_PASSWORD
       valueFrom:
         secretKeyRef:
           name: {{ .Values.dags.gitSync.credentialsSecret | quote }}
-          key: GITSYNC_USERNAME
-    - name: GIT_SYNC_PASSWORD
+          key: GITSYNC_PASSWORD
+      {{- else }}
```

```suggestion
      {{- else if .Values.dags.gitSync.credentialsSecret }}
```
I think this line need to be changed because in the if before, if we have only tag >= 4 and no credentialsSecret it will not enter the if above and will enter this else although it don't have credentialsSecret.